explorer/explorer_pods.go

```diff
@@ -4,33 +4,62 @@ import (
 	"fmt"
 	"strings"
 
+	apiv1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/labels"
 )
 
 const (
 	podLabel = "Pod"
 )
 
 type PodsExplorer struct {
-	Items              []*MenuItem
-	PreviousItem       *MenuItem
-	NamespaceToExplore string
-	Filters            map[string]string
-	PreviousExplorer   Explorable
+	Items                []*MenuItem
+	PreviousItem         *MenuItem
+	NamespaceToExplore   string
+	Filters              map[string]string
+	PreviousExplorer     Explorable
+	PreviousResourceName string
 }
 
 func (n *PodsExplorer) List() error {
 	n.Items = []*MenuItem{}
 
-	// Pods
-	pods, err := k8sclient.CoreV1().Pods(n.NamespaceToExplore).List(v1.ListOptions{
-		LabelSelector: labels.SelectorFromSet(n.Filters).String(),
-	})
+	// List Pods
+	pods, err := k8sclient.Core().Pods(n.NamespaceToExplore).List(v1.ListOptions{})
 	if err != nil {
 		return err
 	}
-	for _, item := range pods.Items {
+
+	var podlist []apiv1.Pod
+	switch n.PreviousExplorer.Kind() {
```

Comment:
Oh I see...

Would it work better to do a type conversion on the interface to see what type it is?